compile.py

```diff
@@ -299,7 +304,9 @@ def add_compliance_matrix(xml_files, adoc_file, has_test_id):
                 requirements = list(sorted(csv.reader(matrix_file)))
                 # requirements is a list, each item of the list has the form
                 # ["requirement name", test_ID]
-                write_matrix_tests(requirements, xml_files, adoc_file)
+                ret = write_matrix_tests(requirements, xml_files, adoc_file)
+                if ret == 1:
```

Comment:
why don't you do directly ?

> return_code =  write_matrix_tests(requirements, xml_files, adoc_file) 

As the function write_matrix_tests is returning 0 or 1

Reply:
write matrix is called multiple time.
Your line of code would take the return code of the last call.
We want it to return 1 if one of the call to write_matrix is 1